### PR TITLE
threadpool, coprocessor: set stack size to 10 MB

### LINKED ISSUE #2428: Use a larger stack size for coprocessor
The coprocessor may have a high recursion which may cause stack overflow, maybe we should use a largerr stack size.

### PR DESCRIPTION
**Pull request by overvenus**
Fix #2428

### CODE REVIEW

File: src/coprocessor/endpoint.rs [hunk 1]
```diff
@@ -171,20 +172,25 @@ impl Host {
             reqs: HashMap::default(),
             last_req_id: 0,
             max_running_task_count: cfg.end_point_max_tasks,
+            // Enpoints may occur a very deep recursion,
+            // so enlarge their stack size to 10 MB, default is 2 MB.
             pool: ThreadPoolBuilder::new(
                 thd_name!("endpoint-normal-pool"),
                 CopContextFactory { sender: r.clone() },
             ).thread_count(cfg.end_point_concurrency)
+                .stack_size(ReadableSize::mb(10).0 as usize)
```
Comment (siddontang) [inline src/coprocessor/endpoint.rs]: define a const var

**Comment by siddontang** [comment]
Rest LGTM

File: src/coprocessor/endpoint.rs (hunk 1 above)
Comment (BusyJay) [reply src/coprocessor/endpoint.rs]: How about make it configurable?

**Comment by BusyJay** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

**Pull request merged**